packages/backend/server/tests/copilot.e2e.ts

```diff
@@ -228,6 +228,61 @@ test('should be able to chat with api', async t => {
   Sinon.restore();
 });
 
+test('should be able to retry with api', async t => {
+  const { app, storage } = t.context;
+
+  Sinon.stub(storage, 'handleRemoteLink').resolvesArg(2);
+
+  // normal chat
+  {
+    const { id } = await createWorkspace(app, token);
+    const sessionId = await createCopilotSession(
+      app,
+      token,
+      id,
+      randomUUID(),
+      promptName
+    );
+    const messageId = await createCopilotMessage(app, token, sessionId);
+    // chat 2 times
+    await chatWithText(app, token, sessionId, messageId);
+    await chatWithText(app, token, sessionId, messageId);
```

Comment:
In the real world case (like in chatgpt), the user may want to **edit** the last message and then resend. I think the current impl does not cover this case?

Reply:
currently pr only cover the case about retry action or latest chat
edit will impl in branching timeline like the chatgpt